### DISCUSSION TITLE
How to return rows if date is null

### DISCUSSION THREAD

**Discussion by HasanAlqaisi** [simolus3/drift#1035] [Q&A] [answered]
Hi,

I want to return rows when the date field is null, i couldn't make this work and I don't know if i'm doing it in the right way..
Here's my code.. date field is (doneAt)

```
return ((select(taskTable)..where((tbl) => tbl.userId.equals(userId)))
        ..where((tbl) => tbl.doneAt.equals(null)))
        .join([
          leftOuterJoin(
              subTaskTable, subTaskTable.taskId.equalsExp(taskTable.id))
        ])
        .watch()
        .map((rows) {
          final result = <TaskData, List<SubTaskData>>{};
          for (final row in rows) {
            final task = row.readTable(taskTable);
            final subtask = row.readTable(subTaskTable);

            final list = result.putIfAbsent(task, () => []);
            if (subtask != null) list.add(subtask);
          }
          return [
            for (final entry in result.entries)
              TaskWithSubtasks(task: entry.key, subtasks: entry.value)
          ];
        });

```

**Comment by simolus3** [comment, marked as answer]
`.equals` generates an `=` in SQL. So if you did `doneAt.equals(null)`, moor would generate `done_at = NULL`. In SQL, comparisons with `NULL` [don't evaluate to a value](https://www.xaprb.com/blog/2006/05/18/why-null-never-compares-false-to-anything-in-sql/), so you can't use that. What you should use is `done_AT IS NULL`, for which you can use the [isNull](https://pub.dev/documentation/moor/latest/moor/isNull.html) function in moor:

```dart
return (select(taskTable)..where((tbl) => tbl.userId.equals(userId) &isNull(tbl.doneAt))
  .join([...])
  // ...
```
